tools/nebula-spark/src/main/scala/com/vesoft/nebula/tools/connector/NebulaOptions.scala

```diff
@@ -0,0 +1,127 @@
+          NebulaOptions.SPACE_NAME     -> spaceName,
+          NebulaOptions.TYPE           -> dataType,
+          NebulaOptions.LABEL          -> label
+        )))
+  }
+
+  /**
+    * Return property with all options
+    * */
+  val asProperties: Properties = {
+    val properties = new Properties()
+    parameters.originalMap.foreach { case (k, v) => properties.setProperty(k, v) }
+    properties
+  }
+
+  /**
+    * parameters invalidate
+    * */
+  // nebula connection info
+  require(parameters.isDefinedAt(HOST_AND_PORTS), s"Option '$HOST_AND_PORTS' is required")
+  val hostAndPorts: String = parameters(HOST_AND_PORTS)
+
+  // nebula space
+  require(parameters.isDefinedAt(SPACE_NAME) && StringUtils.isNotBlank(parameters(SPACE_NAME)),
+          s"Option '$SPACE_NAME' is required and can not be blank")
+  val spaceName: String = parameters(SPACE_NAME)
+
+  // nebula data type: Type.vertex or Type.edge
+  require(parameters.isDefinedAt(TYPE), s"Option '$TYPE' is required")
+  val dataType: String = parameters(TYPE)
+  require(
+    DataTypeEnum.validDataType(dataType),
+    s"Option '$TYPE' is illegal, it should be '${DataTypeEnum.VERTEX}' or '${DataTypeEnum.EDGE}'")
+
+  // nebula label
+  require(parameters.isDefinedAt(LABEL) && StringUtils.isNotBlank(parameters(LABEL)),
+          s"Option '$LABEL' is required and can not be blank")
+  val label: String = parameters(LABEL)
+
+  // nebula return cols
+  private val RETURN_COL_REGEX: String = "(\\w+)(,\\w+)*"
+  require(parameters.isDefinedAt(RETURN_COLS), s"Option '$RETURN_COLS' is required")
+  val returnCols: String = parameters(RETURN_COLS)
+  require(StringUtils.isBlank(returnCols) || Pattern.matches(RETURN_COL_REGEX, returnCols),
+          s"Option '$RETURN_COLS' should be blank or be string like a,b")
+  var allCols: Boolean = false
+
+  // spark partition numbers
+  require(parameters.isDefinedAt(PARTITION_NUMBER), s"Option '$PARTITION_NUMBER' is requied")
+  val partitionNums: String = parameters(PARTITION_NUMBER)
+
+  def getReturnColMap: Map[String, List[String]] = {
+    val result: Map[String, List[String]] = Map()
+    if (StringUtils.isBlank(returnCols)) {
+      allCols = true
+      result ++ Map(label -> List())
+    } else {
+      val properties = returnCols.split(",").toList
+      result ++ Map(label -> properties)
+    }
+  }
+
+  def getHostAndPorts: List[HostAndPort] = {
+    val hostPorts: ListBuffer[HostAndPort] = new ListBuffer[HostAndPort]
+    hostAndPorts
+      .split(",")
+      .foreach(hostPort => { hostPorts.append(HostAndPort.fromString(hostPort)) })
+    hostPorts.toList
+  }
+
+}
+
+class NebulaOptionsInWrite(@transient override val parameters: CaseInsensitiveMap[String])
+    extends NebulaOptions(parameters) {}
+
+object NebulaOptions {
+  private val curId             = new AtomicLong(0L)
+  private val nebulaOptionNames = collection.mutable.Set[String]()
+
+  private def newOption(name: String): String = {
```

Comment:
why return the current options name ？

Reply:
newOption is used to define a variable name，the return is assigned back to option. Actually this method can be moved, and assign the name to variable directly.